modules/react/src/deckgl.js

```diff
@@ -152,25 +164,27 @@ export default class DeckGL extends React.Component {
       // Resolve potentially relative dimensions using the deck.gl container size
       const {x, y, width, height} = viewport;
 
-      // Clone the element with width and height set per view
-      const newProps = Object.assign({}, child.props, {width, height});
-
-      // Inject map properties
-      // TODO - this is too react-map-gl specific
-      Object.assign(newProps, viewport.getMercatorParams(), {
-        visible: viewport.isMapSynched()
-      });
-
-      const clone = cloneElement(child, newProps);
+      let viewChildren = child.props.children;
+      if (typeof viewChildren === 'function') {
+        viewChildren = viewChildren({width, height, viewport, viewState});
+      } else if (Array.isArray(viewChildren)) {
+        viewChildren = viewChildren.map(viewChild => {
+          if (typeof viewChild === 'function') {
+            return viewChild({width, height, viewport, viewState});
```

Comment:
The parameters are the same for all children right? 

We could just create an object ahead of the loop, not sure if it is worth it. 

We could add `x`, `y` to the params as well. They are in viewport I suppose but could make sense to have them accessible.